src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/AttributeCollection.cs

```diff
@@ -19,7 +20,7 @@ public class AttributeCollection : ICollection, IEnumerable
         /// </summary>
         public static readonly AttributeCollection Empty = new AttributeCollection(null);
 
-        private static Hashtable? s_defaultAttributes;
+        private static Dictionary<Type, Attribute?>? s_defaultAttributes;
```

Comment:
Should it be a `ConditionalWeakTable` here as well? Storing `Type`s in a static doctionary would inhibit unloadability. And CWT is thread-safe.

Reply:
I defer to @stephentoub for thoughts here. I was trying to avoid intentionally introducing behavior changes in this PR, so I wasn't intending to make a change like that.

Reply:
They could go in a separate PR but you are already changing this line, better do it the right way immediately. But let's wait for stephentoub.

Reply:
Addressing unrelated, untested concerns around unloadability is orthogonal to this PR.  Let's not conflate the issues.